src/contrib/mlir/lowerer.cpp

```diff
@@ -530,6 +530,102 @@ namespace
         return matchSuccess();
     }
 
+    REWRITER(NGConcatOp)
```

Comment:
Can you please add a unit-test with Concat + other node (e.g. dot/add) ? Unless this is already tested with existing tests.

Reply:
Will do

Reply:
Removing the `compiled_kernel` assert enables two such tests---one of which seems to be failing! (The affine map is not being computed correctly.) Will investigate.

Reply:
(After some offline chatter) looks like this is an unrelated bug, to be addressed by #3233.